.github/workflows/main.yaml

```diff
@@ -0,0 +1,98 @@
+name: CI
+on:
+  push:
+    branches:
+      - 'main'
+      - 'scratch/**'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+.[0-9]+'
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ${{ matrix.config.os }}
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - os: ubuntu-20.04
+            r: 3.6.3
+            # Use pinned URL because later pmxTools requires newer R.
+            rspm: 'https://packagemanager.posit.co/cran/__linux__/focal/2023-02-01'
+          - os: ubuntu-20.04
+            r: 4.0.5
+          - os: ubuntu-20.04
+            r: 4.1.3
+          - os: ubuntu-latest
+            r: release
+    env:
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v4
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+          use-public-rspm: true
+        env:
+          RSPM: ${{ matrix.config.rspm }}
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: |
+            any::devtools
+            any::rcmdcheck
+          upgrade: 'TRUE'
+      - name: Install pdflatex
+        shell: bash
+        run: sudo apt-get install texlive-latex-base texlive-fonts-extra
+      - name: Check package
+        shell: bash
+        run: make ci
+  release:
+    name: Upload release
+    needs: check
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: release
+          use-public-rspm: true
+          # For pkgpub.
+          extra-repositories: 'https://mpn.metworx.com/snapshots/stable/2024-03-01'
```

Comment:
This hard codes the MPN version.  That's not too big of a deal because it's just to get pkgpub, which doesn't change very frequently.  But the plan is to make it more convenient to programmatically determine the latest MPN version so that we can avoid hard coding the MPN versions in spots like this.

(At some point, we may try to pull much of this release step into shared actions that can just be referenced by workflows across metrumresearchgroup repos.)

Reply:
`mpn.metworx.com` now advertises latest URL.  Updated to query.